unity-client/Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/ChatHUDController.cs

```diff
@@ -0,0 +1,67 @@
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine.Events;
+using ChatMessage = ChatController.ChatMessage;
+using ChatMessageType = ChatController.ChatMessageType;
+public class ChatHUDController : IDisposable
+{
+    public const int MAX_CHAT_ENTRIES = 100;
+
+    public ChatHUDView view;
+
+    public UnityAction<string> OnSendMessage;
+
+    public void Initialize(ChatHUDView view = null, UnityAction<string> onSendMessage = null)
+    {
+        if (view == null)
+        {
+            this.view = ChatHUDView.Create();
+        }
+        else
+        {
+            this.view = view;
+        }
+
+        this.view.Initialize(this, onSendMessage);
+    }
+
+    public void AddChatMessage(ChatMessage message)
+    {
+        view.AddEntry(message);
+
+        if (view.entries.Count > MAX_CHAT_ENTRIES)
+        {
+            var result = TrimAndSortChatMessages(view.entries.Select((x) => x.message).ToList());
+            view.RepopulateAllChatMessages(result);
+        }
+    }
+
+    public void FilterByType(ChatMessageType type)
+    {
+        var result = ChatController.i.entries.Where((x) => x.messageType == type).ToList();
+        result = TrimAndSortChatMessages(result);
+        view.RepopulateAllChatMessages(result);
+    }
+
+    public List<ChatMessage> TrimAndSortChatMessages(List<ChatMessage> messages)
+    {
+        var result = messages;
+
+        result.Sort((x, y) => { return x.timestamp > y.timestamp ? 1 : -1; });
+
+        if (result.Count > MAX_CHAT_ENTRIES)
+        {
+            int entriesToRemove = (result.Count - MAX_CHAT_ENTRIES);
+            result.RemoveRange(0, entriesToRemove);
+        }
```

Comment:
I think instead of re-sorting and removing entries, every time we get to the MAX_CHAT_ENTRIES we could just remove the oldest message in the list/queue, reconfigure it, and re-add it at the back of the list/queue again and that would be it. (maybe using a queue instead of a list could make it easier as well)

Reply:
This is a good idea, i'm going to give it a shot.